### ISSUE TITLE
Unexpected missing-argument diagnostic with pyspark

### ISSUE THREAD

**Issue by KaufmanDmitriy** [astral-sh/ty#3586]
### Summary

Code example:
```python
from pyspark.sql import SparkSession, functions as F

spark = (
    SparkSession.builder
    .appName("whatever")
    .getOrCreate()
)

df = spark.read.table("test")

df = df.filter(F.col("id").isNotNull())
```

After using ty with command
```shell
uvx ty check
```

I get this diagnostic

```markdown
error[missing-argument]: No argument provided for required parameter 1
  --> main.py:11:16
   |
11 | df = df.filter(F.col("id").isNotNull())
   |                ^^^^^^^^^^^^^^^^^^^^^^^
   |
```

Versions used:
pyspark==3.5.1
ty 0.0.40 (7b95bc219 2026-05-27)

### Version

ty 0.0.40 (7b95bc219 2026-05-27)

**Comment by carljm** [comment]
Thanks for the report! This is #491. In older versions of pyspark (prior to https://github.com/apache/spark/commit/e01ac581f46aa595e66daf33fe92b56d1328bc78), methods like `isNotNull` are [defined by calling a `_unary_op` helper](https://github.com/apache/spark/blob/branch-3.5/python/pyspark/sql/column.py#L1110) which [returns a `Callable[[Column], Column]`](https://github.com/apache/spark/blob/branch-3.5/python/pyspark/sql/column.py#L113). It's fundamentally ambiguous whether a `Callable` type like this should be assumed to act like a bound-method descriptor or not: some callables (like functions) do, and many other callables (like instances with a `__class__` method) may not. Different type checkers have different heuristics for this unfortunate ambiguity.

More recent versions of pyspark no longer use this `_unary_op` helper, and so this problem doesn't occur.
